shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -51,48 +63,145 @@ public PlatformViewRegistry getRegistry() {
     }
 
     public void onFlutterViewDestroyed() {
-        // TODO(amirh): tear down all vd resources.
+        for (int id : vdControllers.keySet()) {
+            VirtualDisplayController controller = vdControllers.get(id);
+            controller.dispose();
+        }
+        vdControllers.clear();
     }
 
     @Override
-    public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+    public void onMethodCall(final MethodCall call, final MethodChannel.Result result) {
+        if (Build.VERSION.SDK_INT < MINIMAL_SDK) {
+            Log.e(TAG, "Trying to use platform views with API " + Build.VERSION.SDK_INT
+                    + ", required API level is: " + MINIMAL_SDK);
+            return;
+        }
         switch (call.method) {
             case "create":
-                createPlatformView(call);
-                break;
+                mFlutterView.post(new Runnable() {
```

Comment:
These messages should be received on the platform thread.  Does something fail if you call createPlatformView directly here?

Reply:
done, thanks